### PR TITLE
EQW calculations

### PR DESCRIPTION
**Pull request by alexmaragko**
This PR adds the equivalent width calculation of the separate components, and those of the combined dust features, and should replace PR #143.

An updated output table is also provided in `fit_spectrum.rst`.

Differences with PR #143:
- The `eqws` method is now part of the `feature_strengths` module.
- The PR #143 comments are addressed.

### CODE REVIEW

File: pahfit/base.py [hunk 1]
```diff
@@ -311,7 +311,8 @@ def plot(self, axs, x, y, yerr, model, scalefac_resid=2):
             if isinstance(cmodel, BlackBody1D):
                 cont_components.append(cmodel)
                 # plot as we go
-                ax.plot(x, cmodel(x) * ext_model / x, "#FFB000", alpha=0.5)
+                # ax.plot(x, cmodel(x) * ext_model / x, "#FFB000", alpha=0.5)
+                ax.plot(x, cmodel(x) * ext_model / x, "limegreen", alpha=0.5)
```
Comment (els1) [inline pahfit/base.py:315]: not clear why line #314 is uncommented and line #315 is added which seems identical aside from the color? 
BTW if you changes the plotting colors, the docs may need to be updated.

File: pahfit/base.py [hunk 2]
```diff
@@ -320,11 +321,13 @@ def plot(self, axs, x, y, yerr, model, scalefac_resid=2):
         # now plot the dust and gas lines
         for cmodel in model:
             if isinstance(cmodel, Gaussian1D):
+                # ax.plot(x, (cont_y + cmodel(x)) * ext_model / x, "#DC267F", alpha=0.5)
```
Comment (els1) [inline pahfit/base.py:324]: Why is this added?

File: pahfit/base.py [hunk 3]
```diff
@@ -495,6 +511,7 @@ def save(obs_fit, filename, outform):
                         component.fwhm.fixed,
                         strength,
                         strength_unc,
+                        eqw
                     ]
                 )
             elif comp_type == "Gaussian1D":
```
Comment (els1) [inline pahfit/base.py:517]: add below this line comment:
# Calculate feature strength

File: pahfit/feature_strengths.py [hunk 4]
```diff
@@ -103,6 +109,79 @@ def featcombine(ftable):
...
+                        fwhm=fwhm)
+        lnu = drude(lam)
+
+    elif comp_type == 'Gaussian1D':
+        gauss = Gaussian1D(amplitude=amp,
+                           mean=x_0,
+                           stddev=fwhm_stddev)
+        lnu = gauss(lam)
+
+    # Set broad limit (bl) to replace the continuum in the EQW calculation
```
Comment (els1) [inline pahfit/feature_strengths.py:172]: Perhaps add:
# Following the EQW calculation in IDL PAHFIT, we define a default broad limit bl.

**Review by els1 — CHANGES_REQUESTED** [review]

File: pahfit/feature_strengths.py [hunk 5]
```diff
@@ -103,6 +109,79 @@ def featcombine(ftable):
...
+    if comp_type == 'Gaussian1D':
+        fwhm = 2 * fwhm_stddev * np.sqrt(2 * np.log(2))
+    else:
+        fwhm = fwhm_stddev
+
+    # Get range and wavelength region for integration.
+    low = x_0 - (fwhm * 6)
+    lmin = low if low > 0 else 0.
+    lmax = x_0 + (fwhm * 6)
+    lam = np.arange(100) / 99 * (lmax - lmin) + lmin
```
Comment (Ameek-Sidhu) [inline pahfit/feature_strengths.py:148]: This line is slightly confusing to me. lam = np.linspace(lmin, max, num=100) would also produce the same result. I find np.linspace will be more intuitive than np.arange in this case.

File: pahfit/base.py (hunk 1 above)
Comment (alexmaragko) [reply pahfit/base.py:315]: Sorry this is an artifact from another branch I made based on this. Not sure why it made it here. I will fix this.

File: pahfit/base.py (hunk 2 above)
Comment (alexmaragko) [reply pahfit/base.py:324]: Same as the above comment. Will be removed.

File: pahfit/feature_strengths.py [hunk 6]
```diff
@@ -103,6 +109,79 @@ def featcombine(ftable):
...
+                         np.sum(dfs[mask]['eqw'])
+                         ])
 
     return cftable
+
+
+def eqws(comp_type, x_0, amp, fwhm_stddev, obs_fit):
+    """
+    Calculate the emission features equivalent width
+    (integral[(I_nu-I_cont)/I_cont d_lam]) in microns.
```
[the remark this answers is not in the document]
Comment (alexmaragko) [reply pahfit/feature_strengths.py:121]: This comment was added per Els's request in PR #143.

File: pahfit/feature_strengths.py (hunk 5 above)
Comment (alexmaragko) [reply pahfit/feature_strengths.py:148]: Agreed, it's the same thing. I will change it.

**Comment by karllark** [comment]
Is this PR ready to merge?  Or is more review needed?

**Comment by alexmaragko** [comment]
> Is this PR ready to merge? Or is more review needed?

From my end it is ready. I have addressed the raised comments.

**Comment by karllark** [comment]
@els1 : ok to merge?

**Comment by karllark** [comment]
@Ameek-Sidhu and @els : ok for merging?

**Comment by Ameek-Sidhu** [comment]
It is ready to merge from my perspective.

**Comment by els1** [comment]
Good to merge.
